File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    type: str
    title: str
    message: str
    is_read: bool
    link: Optional[str] = None
    related_id: Optional[str] = None
    related_type: Optional[str] = None
    created_at: datetime


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    unread_count: int
    has_more: bool


class NotificationUnreadCountResponse(BaseModel):
    unread_count: int

File: frontend/src/types/notification.ts
export interface NotificationItem {
  id: string;
  type: string;
  title: string;
  message: string;
  is_read: boolean;
  link: string | null;
  related_id: string | null;
  related_type: string | null;
  created_at: string;
}

export interface NotificationListResponse {
  items: NotificationItem[];
  unread_count: number;
  has_more: boolean;
}
